### PR TITLE
Opt-in automatic forking for AndcultureCode repos

### PR DESCRIPTION
**Pull request by wintondeshong**
If enabled (via `-f|--fork` flag), workspace command will guide the user through configuring their github auth and then automatically create forks for each of the mainline AndcultureCode repos.

From there, being we have auto cloning of the workspace, it proceeds to then automatically clone those new forks into their workspace. 

Marry this with @brandongregoryscott new custom project CLIs, if one were to setup a custom cli at the top of their workspace, they could make their own workspace a global plugin in and of itself... for easy management (just sayin')

-   [x] Related GitHub issue(s) linked in PR description
-   [x] Destination branch merged, built and tested with your changes
-   [x] Code formatted and follows best practices and patterns
-   [x] Code builds cleanly (no _additional_ warnings or errors)
-   [x] Manually tested
-   [x] Automated tests are passing
-   [x] No _decreases_ in automated test coverage
-   [x] Documentation updated (readme, docs, comments, etc.)
-   [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Comment by mrjones2014** [comment]
**Suggestion:** We could probably also easily add the upstream repo as a secondary `upstream` remote after creating the fork, so that, without the user doing any further configuration, they could pull upstream changes by running `git pull upstream [branch]`

**Comment by wintondeshong** [comment]
> **Suggestion:** We could probably also easily add the upstream repo as a secondary `upstream` remote after creating the fork, so that, without the user doing any further configuration, they could pull upstream changes by running `git pull upstream [branch]`

Definitely agree! Already in the plans. If there are any other helpful config items off the top of your head, let me know.

**Review by mrjones2014 — APPROVED** [review]

File: and-cli-workspace.js [hunk 1]
```diff
@@ -64,19 +74,72 @@ require("./command-runner").run(async () => {
...
+
+        // Additional configuration if its a clone of a fork
+        if (!repo.fork) {
+            return;
+        }
+
+        const folder = git.getCloneDirectoryName(repo.name, prefix);
+        shell.cd(folder);
+
+        const addUpgradeResult = git.addRemote(
```
Comment (brandongregoryscott) [inline and-cli-workspace.js:97]: Question - is this meant to read `addUpstreamResult`?

File: modules/github.js [hunk 2]
```diff
@@ -6,33 +6,53 @@ const { createNetrcAuth } = require("octokit-auth-netrc");
...
 const fs = require("fs");
 const git = require("./git");
+const js = require("./js");
 const { Octokit } = require("@octokit/rest");
 const os = require("os");
 const upath = require("upath");
 const userPrompt = require("./user-prompt");
+const { StringUtils } = require("andculturecode-javascript-core");
+const { fork } = require("child_process");
+const { sleep } = require("./js");
```
Comment (brandongregoryscott) [inline modules/github.js:16]: Nitpick - in terms of alphabetization, I've been trying to keep the destructured imports above the standard module imports. What do you think?

File: modules/js.js [hunk 3]
```diff
@@ -22,6 +22,69 @@ const js = {
             await callback(array[index], index, array);
         }
     },
+
+    /**
+     * Blocks for a given duration
+     * @param {number} milliseconds number of milli-seconds to wait
```
Comment (brandongregoryscott) [inline modules/js.js:28]: I _believe_ we have a `sleep` function with a very similar/identical implementation in `CoreUtils` from andculturecode-javascript-core - but double check

**Review by brandongregoryscott — APPROVED** [review]
Some minor refactoring food for thought - looks great! Excited to have this to easily manage our OSS.

File: and-cli-workspace.js (hunk 1 above)
Comment (wintondeshong) [reply and-cli-workspace.js:97]: yup, who knows why i typed 'upgrade' 🤷

File: modules/github.js (hunk 2 above)
Comment (wintondeshong) [reply modules/github.js:16]: Yeah, I'm down for that. Definitely helps grouping them

File: modules/js.js (hunk 3 above)
Comment (wintondeshong) [reply modules/js.js:28]: doh 🤦 - blame it on being on vacation

**Pull request merged**